demo/src/common/DemoTemplate.tsx

```diff
@@ -50,60 +51,62 @@ export const DemoTemplate = (props: DemoTemplateProps) => {
   };
   return (
     <ThemeContext.Provider value={{ theme, setTheme }}>
-      <InformationPanelWrapper>
-        <div className='demo-template-container'>
-          <div className='demo-template-content'>
-            {children}
-            {!isTestRun() && (
-              <ButtonGroup className='demo-template-button-overlay'>
-                <ThemeButton />
-                <IconButton onClick={() => setShowCodeDemo((f) => !f)}>
-                  <SvgDeveloper />
-                </IconButton>
-              </ButtonGroup>
-            )}
-          </div>
+      <ThemeProvider theme={theme}>
```

Comment:
There should only be one ThemeProvider.

either remove this or remove the one in `main.tsx`

Reply:
now the styles on the home page are broken

![](https://github.com/iTwin/iTwinUI-layouts/assets/9084735/ff440d73-cf9e-4a20-b84b-73b0bacf4c9c)


probably would be better to do it the other way around. you can also move the context up if that helps

Reply:
well yeah, you can't just "remove" it. you need to *move* the ThemeProvider from `DemoTemplate.tsx` up to `main.tsx`, and along with it you'll also need to move the ThemeContext so that switching themes continues to work.

the ThemeContext value will be used in DemoTemplate to switch themes.

Reply:
you'll need to pass the same thing that was previously passed to the old `ThemeContext.Provider`, i.e. the theme state.

in other words, you need to "lift" the theme state up so that it can be used by both `ThemeProvider` and the `ThemeButton`.

have you read through the react docs? it explains everything in detail.
https://react.dev/learn/sharing-state-between-components
https://react.dev/learn/passing-data-deeply-with-context